beacon-chain/sync/subscriber_data_column_sidecar.go

```diff
@@ -52,3 +60,68 @@ func (s *Service) receiveDataColumnSidecar(ctx context.Context, sidecar blocks.V
 
 	return nil
 }
+
+// triggerGetBlobsV2ForDataColumnSidecar triggers getBlobsV2 retry when receiving a data column sidecar.
+// This function attempts to fetch the block and trigger the execution service's retry mechanism.
+func (s *Service) triggerGetBlobsV2ForDataColumnSidecar(ctx context.Context, blockRoot [32]byte) error {
+	// Check if service is properly configured
+	if s.cfg == nil || s.cfg.chain == nil || s.cfg.beaconDB == nil {
+		log.Debug("Service not properly configured for getBlobsV2 retry trigger")
+		return nil
+	}
+
+	// Try to get the block from the database or cache
+	if !s.cfg.chain.HasBlock(ctx, blockRoot) {
+		// Block not available yet, this is expected in some cases
+		log.WithField("blockRoot", fmt.Sprintf("%#x", blockRoot)).Debug("Block not available for getBlobsV2 retry trigger")
+		return nil
+	}
+
+	// Get the specific block by root from database
+	signedBlock, err := s.cfg.beaconDB.Block(ctx, blockRoot)
+	if err != nil {
+		log.WithError(err).Debug("Could not fetch block from database for getBlobsV2 retry trigger")
+		return nil
+	}
+	if signedBlock == nil || signedBlock.IsNil() {
+		log.Debug("Block not found in database for getBlobsV2 retry trigger")
+		return nil
+	}
+
+	// Check if this block has blob commitments that would need getBlobsV2
+	blockBody := signedBlock.Block().Body()
+	commitments, err := blockBody.BlobKzgCommitments()
+	if err != nil {
+		return err
+	}
+	if len(commitments) == 0 {
+		return nil
+	}
+
+	// Trigger the retry by calling the execution service's reconstruct method
+	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				log.WithField("error", r).
+					WithField("recoveredAt", "triggerGetBlobsV2ForDataColumnSidecar").
+					WithField("stack", string(debug.Stack())).
+					Error("Panic occurred in getBlobsV2 retry goroutine")
+			}
+		}()
+
+		bgCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+
+		log.WithField("blockRoot", fmt.Sprintf("%#x", blockRoot)).Debug("Triggering getBlobsV2 retry for data column sidecar")
+
+		// This will trigger retry logic if needed
+		if s.cfg.executionReconstructor != nil {
```

Comment:
Better to do the opposite.
Return if nil.
The continue the normal execution if not nil.